ironfish-rust/src/serializing/fr.rs

```diff
@@ -0,0 +1,109 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use crate::errors::{IronfishError, IronfishErrorKind};
+use jubjub::Fr;
+
+use super::{bytes_to_hex, hex_to_bytes};
+
+pub const MAC_SIZE: usize = 16;
+
+pub trait FrSerializable {
+    fn serialize(&self) -> [u8; 32];
+    fn deserialize(bytes: [u8; 32]) -> Result<Fr, IronfishError>;
+    fn hex_key(&self) -> String;
+    fn from_hex(hex_key: &str) -> Result<Fr, IronfishError>;
+}
+
+impl FrSerializable for Fr {
+    fn serialize(&self) -> [u8; 32] {
+        self.to_bytes()
+    }
+
+    fn deserialize(bytes: [u8; 32]) -> Result<Self, IronfishError> {
+        let nsk = match Fr::from_bytes(&bytes).into() {
+            Some(nsk) => nsk,
+            None => {
+                return Err(IronfishError::new(
+                    IronfishErrorKind::InvalidProofAuthorizingKey,
```

Comment:
`Fr` isn not just a proof authorizing key, it is a scalar field element, the proof authorizing key just happens to be a scalar field element.